docs/okeps/okep-5224-connecting-udns/okep-5224-connecting-udns.md

```diff
@@ -0,0 +1,1275 @@
+---
+sh-5.2# ovn-nbctl show blue_blue.network_ovn_cluster_router
+router 13b24c4d-f3c3-4811-98a9-98cc996be0d4 (blue_blue.network_ovn_cluster_router)
+    port blue_blue-network-ovn-worker-to-colored-enterprise
+        mac: "0a:58:c0:80:00:06"
+        ipv6-lla: "fe80::858:c0ff:fe80:6"
+        networks: ["192.128.0.6/31"]
+---
+_uuid               : 24aabdb9-0b37-4332-965d-5d09ae0694a8
+dhcp_relay          : []
+enabled             : []
+external_ids        : {}
+gateway_chassis     : []
+ha_chassis_group    : []
+ipv6_prefix         : []
+ipv6_ra_configs     : {}
+mac                 : "0a:58:c0:80:00:06"
+name                : blue_blue-network-ovn-worker-to-colored-enterprise
+networks            : ["192.128.0.6/31"]
+options             : {}
+peer                : colored-enterprise-to-blue_blue-network-ovn-worker --> connects to connect-router
+status              : {}
+---
+sh-5.2# ovn-nbctl show green_green.network_ovn_cluster_router
+router 15532710-213f-4bee-8bc1-9d500a6bc5a9 (green_green.network_ovn_cluster_router)
+    port green_green-network-ovn-worker-to-colored-enterprise
+        mac: "0a:58:c0:80:00:0a"
+        ipv6-lla: "fe80::858:c0ff:fe80:a"
+        networks: ["192.128.0.10/31"]
+---
+_uuid               : 99c4bff1-17c9-4d17-826c-c069f2a3cb6b
+dhcp_relay          : []
+enabled             : []
+external_ids        : {}
+gateway_chassis     : []
+ha_chassis_group    : []
+ipv6_prefix         : []
+ipv6_ra_configs     : {}
+mac                 : "0a:58:c0:80:00:0a"
+name                : green_green-network-ovn-worker-to-colored-enterprise
+networks            : ["192.128.0.10/31"]
+options             : {}
+peer                : colored-enterprise-to-green_green-network-ovn-worker
+status              : {}
+```
+
+**Connecting Layer2 type networks together**
+
+The below diagram shows the overlay part of the OVN Topology that
+OVN-Kubernetes creates for 3 UDN networks (blue, green and yellow)
+across two nodes. NOTE: The topology representation is that of the new
+upcoming Layer2 topology. The dotted pieces in network colors represent
+the bits in design and not there yet. See enhancement on [new Layer2
+topology] for more details on the new design.
+
+[new Layer2 topology]: https://github.com/ovn-kubernetes/ovn-kubernetes/pull/5097
+
+![3-isolated-l2-networks](https://raw.githubusercontent.com/tssurya/ovn-kubernetes/connecting-udns-okep/docs/okeps/okep-5224-connecting-udns/images/l2-connecting-udns-0.png)
+
+**Add a new transit router between the transit routers of the networks**
+
+When the `ClusterNetworkConnect` CR gets created selecting these three
+networks, the controller will make the following changes to the topology:
+* Create a distributed `connect-router-colored-enterprise` (basically a
+  transit-router type in OVN used for interconnection)
+* Connect the `connect-router-colored-enterprise` to the `transit-router`’s
+  of the blue, green and yellow networks on each node using patch ports and
```

Comment:
So we use <number of nodes> links to connect 2 transit routers, is it to ensure symmetric reply? or why not just use 1 link?

Reply:
@tssurya yes, i was referring to the interconnect router when connecting L2 <--> L2.  It's not a transit router (i.e., it has no "remote" ports in none of the NB representations).